src/module.c

```diff
@@ -6074,6 +6074,15 @@ const char *moduleTypeModuleName(moduleType *mt) {
     return mt->module->name;
 }
 
+/* Return the module name from command, NULL if the command is not a module command */
+const char *moduleNameFromCommand(struct redisCommand *cmd) {
+    if (cmd->proc != RedisModuleCommandDispatcher)
+        return NULL;
```

Comment:
Can this ever happen? If not, maybe an assert is better?

Reply:
no strong feelings one way or the other... @oranagra please decide

Reply:
since we don't use `cmd->proc` in this code, i don't think we need it at all.
unless we consider this check a safeguard against the casting of `cmd->getkeys_proc`?
in which case, i guess an assert is better.

p.s. i really hate that silly trick, maybe it's time to sort that out and add a proper module member....

Reply:
>p.s. i really hate that silly trick, maybe it's time to sort that out and add a proper module member....

agree, `proc` and `getkeys_proc` is ambiguous and silly

Reply:
https://github.com/redis/redis/issues/10552